### PR TITLE
[stable/datadog] introduce dnsConfig helm values for the different co…

### PR DESCRIPTION
**Pull request by orong-pp**
…ntainers

Signed-off-by: Oron Gola <oron.gola@gmail.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by orong-pp** [comment]
looks like I'm hitting the same issue like here #16070 
@vboulineau

**Comment by orong-pp** [comment]
@CharlyF please review

**Comment by orong-pp** [comment]
### some details:
**the issue:**
when kubernetes runs a dns query in one of its pods, it first tries to look for the record in its internal namespaces. it can take 4-5 queries that search different options before we actually get to the final answer
in case we query an external website like spot.io we get 4-5 times more queries than we actually need and this puts unwanted load on coredns pods.

**the details for this are all explained here**
https://dev.to/mrkaran/dns-lookups-in-kubernetes-5cm1

**solution:**
so to deal with this issue one can define the ndots dns configuration option
the default is 5 and it means that only urls with 5 or more dots (.) in it will be queried directly from external dns and not go through 4 redundant dns queries

to set the ndots config option lower I added this pull request and made it possible to control this value and set it lower, so that all queries are direct and no internal search is made

**dnsConfig documentation:**
this is how dnsConfig should be configured according to kubernetes docs
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/

@vboulineau please review

File: stable/datadog/values.yaml [hunk 1]
```diff
@@ -540,6 +540,14 @@ clusterAgent:
   #
   useHostNetwork:  # true
 
+  ## @param dnsConfig - list of objects - optional
+  ## specify dns configuration options for datadog cluster agent containers e.g ndots
+  dnsConfig:
+  #  dnsConfig:
```
Comment (vboulineau) [inline stable/datadog/values.yaml]: This indented `# dnsConfig` line is misleading, some people may think they need to double it.
Also as this field content should be a valid Kubernetes dnsConfig, you should link the Kubernetes doc:
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config

Comment (orong-pp) [reply stable/datadog/values.yaml]: thanks for the review @vboulineau 
I agree and I changed my docs and suggested default value as you requested

please review again

**Comment by heydonovan** [comment]
Can we get these changes added to the `CHANGELOG.md`? :v:

**Pull request merged**